test/functional/test_framework/util.py

```diff
@@ -390,15 +390,63 @@ def sync_blocks(rpc_connections, *, wait=1, timeout=60):
         time.sleep(wait)
     raise AssertionError("Block sync timed out:{}".format("".join("\n  {!r}".format(b) for b in best_hash)))
 
-def sync_mempools(rpc_connections, *, wait=1, timeout=60, flush_scheduler=True):
+def sync_mempools(rpc_connections, *, wait=1, timeout=60, use_rpc_sync=False, flush_scheduler=True):
     """
     Wait until everybody has the same transactions in their memory
-    pools
+    pools. If use_rpc_sync is set, sync all transactions right away.
     """
+
+    class TxInfo:
+        def __init__(self, *, raw, ancestors):
+            self.raw = raw
+            self.ancestors = ancestors
+
+    def topo_send(txs, rpc, pool_add):
+        for i in txs:
+            topo_send(txs[i].ancestors, rpc, pool_add)
+            if i not in pool_add:
+                try:
+                    assert_equal(i, rpc.sendrawtransaction(txs[i].raw))
+                    pool_add.add(i)
+                    # Note that conflicted txs (due to RBF) are not removed
+                    # from the pool
+                except JSONRPCException as e:
+                    # This transaction violates policy (e.g. RBF policy). The
+                    # mempools should still converge when the high-fee
+                    # replacement is synced in a later call
+                    assert 'insufficient fee' in e.error['message']
+
     stop_time = time.time() + timeout
     while time.time() <= stop_time:
         pool = [set(r.getrawmempool()) for r in rpc_connections]
-        if pool.count(pool[0]) == len(rpc_connections):
+        sync_done = pool.count(pool[0]) == len(rpc_connections)
+        if use_rpc_sync and not sync_done:
+            # Iterate over all nodes, get their raw mempool and send the
+            # missing txs to all other nodes
+            for i_remote, rpc_remote in enumerate(rpc_connections):
```

Comment:
nit: `remote` and `target` don't seem like very descriptive names to me `from` and `to` seem more intuitive to me as the node you're sync'ing from and to.

Reply:
Renamed to `to` and `from`